compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -369,6 +369,10 @@ impl<'tcx> Printer<'tcx> for SymbolMangler<'tcx> {
                 ty.print(self)?;
             }
 
+            ty::Pat(ty, _pat) => {
+                ty.print(self)?;
```

Comment:
we should mangle these differently -- won't we get symbol conflicts if not? this feels relevant since we still emit symbols for things like like builtin impl fns.

Reply:
done and added tests